app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -409,9 +413,26 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         }
 
         if (!browserOverlay.isVisible && session.isYoutubeTV &&
-                event.keyCode == KeyEvent.KEYCODE_BACK) {
-            val escKeyEvent = KeyEvent(event.action, KeyEvent.KEYCODE_ESCAPE)
-            (activity as MainActivity).dispatchKeyEvent(escKeyEvent)
+                event.keyCode == KeyEvent.KEYCODE_BACK &&
+                event.action == KeyEvent.ACTION_DOWN) {
+            webView?.evalJS("""
+                (function () {
+                    return document.activeElement.parentElement.parentElement.id === 'guide-list';
+                })();
+                """,
+                    ValueCallback {
+                        val keyCode = if (it == "true") {
+                            val wv = (webView as ViewGroup).getChildAt(0) as WebView
+                            val backForward = wv.copyBackForwardList().toStringList()
+                            val youtubeIndex = backForward.lastIndexOf("https://ftv.cdn.mozilla.net/ytht")
+                            val goBackSteps = backForward.size - youtubeIndex
+                            wv.goBackOrForward(-goBackSteps)
+                            KeyEvent.KEYCODE_BACK
```

Comment:
Do we need to dispatch this back press at all if we're already going back before YouTube? I don't think so but I could be mistaken.